malaria24/ona/models.py

```diff
@@ -228,29 +233,44 @@ def compile_digest(cls):
         digest.save()
         return digest
 
-    def get_digest_email_data(self, facility_code):
+    def get_digest_email_data(self, district, facility_code):
         date = datetime.today()
         week = 'Week ' + str(date.strftime("%U")) + ' ' + str(date.year)
-        facility_cases = ReportedCase.objects.filter(
-            facility_code=facility_code, digest__isnull=True)
-        females = facility_cases.filter(gender__icontains='f').count()
-        males = facility_cases.exclude(gender__icontains='f').count()
-        over5 = len([x for x in facility_cases if x.age >= 5])
-        under5 = facility_cases.count() - over5
-
-        facilities = Facility.objects.filter(facility_code=facility_code)
-        if facilities.exists():
-            facility_name = facilities.first().facility_name
-        else:
-            facility_name = 'Unknown (facility code: %s)' % (facility_code,)
-
-        fac_list = [{
-            'facility': facility_name,
-            'cases': facility_cases.count(),
-            'females': females, 'males': males,
-            'under5': under5,
-            'over5': over5,
-            'week': week}]
+
+        if not district:
+            try:
+                district = Facility.objects.get(
+                    facility_code=facility_code).district
+            except:
+                return {}
+
+        district_fac_codes = Facility.objects.filter(
+            district=district).values_list(
+                'facility_code',
+                flat=True).distinct().order_by("district")
+
+        district_cases = ReportedCase.objects.filter(
+            facility_code__in=district_fac_codes, digest__isnull=True)
+        females = district_cases.filter(gender__icontains='f').count()
+        males = district_cases.exclude(gender__icontains='f').count()
+        over5 = len([x for x in district_cases if x.age >= 5])
+        under5 = district_cases.count() - over5
+
+        facilities = Facility.objects.filter(district=district)
+        fac_list = []
+        for fac in facilities:
+            if fac:
+                facility_name = fac.facility_name
+            else:
+                facility_name = 'Unknown (district: %s)' % (district,)
```

Comment:
You don't have to change this, but there's a neat python shorthand that looks like this
```python
facility_name = fac.facility_name if facility_name else  'Unknown (district: %s)' % (district,)
```